packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -453,44 +453,95 @@ class CupertinoPageTransition extends StatelessWidget {
 ///
 /// For example, used when creating a new calendar event by bringing in the next
 /// screen from the bottom.
-class CupertinoFullscreenDialogTransition extends StatelessWidget {
+class CupertinoFullscreenDialogTransition extends StatefulWidget {
   /// Creates an iOS-style transition used for summoning fullscreen dialogs.
   ///
+  const CupertinoFullscreenDialogTransition({
+    super.key,
+    required this.primaryRouteAnimation,
+    required this.secondaryRouteAnimation,
+    required this.child,
+    required this.linearTransition,
+  });
+
+
   ///  * `primaryRouteAnimation` is a linear route animation from 0.0 to 1.0
   ///    when this screen is being pushed.
+  final Animation<double> primaryRouteAnimation;
+
   ///  * `secondaryRouteAnimation` is a linear route animation from 0.0 to 1.0
   ///    when another screen is being pushed on top of this one.
-  ///  * `linearTransition` is whether to perform the secondary transition linearly.
+  final Animation<double> secondaryRouteAnimation;
+
+  ///  * `linearTransition` is whether to perform the transitions linearly.
   ///    Used to precisely track back gesture drags.
-  CupertinoFullscreenDialogTransition({
-    super.key,
-    required Animation<double> primaryRouteAnimation,
-    required Animation<double> secondaryRouteAnimation,
-    required this.child,
-    required bool linearTransition,
-  }) : _positionAnimation = CurvedAnimation(
-         parent: primaryRouteAnimation,
+  final bool linearTransition;
+
+  /// The widget below this widget in the tree.
+  final Widget child;
+
+  @override
+  State<CupertinoFullscreenDialogTransition> createState() => _CupertinoFullscreenDialogTransitionState();
+}
+
+class _CupertinoFullscreenDialogTransitionState extends State<CupertinoFullscreenDialogTransition> {
+
+    // When this page is coming in to cover another page.
+  late Animation<Offset> _primaryPositionAnimation;
+  // When this page is becoming covered by another page.
+  late Animation<Offset> _secondaryPositionAnimation;
+  // Curve of primary page which is coming in to cover another page.
+  CurvedAnimation? _primaryPositionCurve;
+  // Curve of secondary page which is becoming covered by another page.
+  CurvedAnimation? _secondaryPositionCurve;
+
+  @override
+  void initState() {
+    super.initState();
+    _setupAnimation();
+  }
+
+  @override
+  void didUpdateWidget(covariant CupertinoFullscreenDialogTransition oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.primaryRouteAnimation != widget.primaryRouteAnimation
+    || oldWidget.secondaryRouteAnimation != widget.secondaryRouteAnimation
+    || oldWidget.child != widget.child || oldWidget.linearTransition != widget.linearTransition) {
```

Comment:
I think you need to remove `oldWidget.child != widget.child`, this is not used to create the animation

Reply:
@Dimilkalathiya , can you respond this message? Why we recalculate animations if child is changed?

Reply:
@polina-c  if `oldWidget.child != widget.child` not there it changes behaviour of animation offset i am not entirely sure is to why this is happening first i thought it might be performance improvement, but offsets are quite far to consider just that, so i decided to let it be there since previous implementation was also rebuilding widgets, would that be a problem?

Reply:
Ok, thanks, let it stay there.

Reply:
Isn't it the same as what is happening in my PR ?